src/main/java/org/cirdles/topsoil/app/utils/SVGSaver.java

```diff
@@ -29,15 +30,45 @@
 import org.w3c.dom.Document;
 
 /**
- *
+ * The purpose of SVGSaver.java is to save a vector image displayed in Topsoil's
+ * js-charts with the .svg file extension
+ * 
  * @author John Zeringue
  */
+
 public class SVGSaver {
 
     public static final String SVG_DOCTYPE_PUBLIC = "-//W3C//DTD SVG 1.1//EN";
     private static final String SVG_DOCTYPE_SYSTEM = "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd";
 
+    /**
+     * Saves the vector image in Topsoil with the .svg file extension
+     * 
+     * @param svgDocument
+     *            Vector image to be saved
+     */
     public void save(Document svgDocument) {
+
+        File svgFile = generateSaveUI();
+        Optional<File> svgOption = Optional.ofNullable(svgFile);
+
+        // If there is no file specified, then there is nothing to return.
+        // Otherwise, you would take that document and write it to that
+        // specified file.       
+        if(svgOption.isPresent()) {
+            writeSVGToFile(svgDocument, svgFile);
+        }
+        else {
+            return;
+        }
```

Comment:
You don't quite understand `Optional`.

This whole method body should look something like:

``` java
generateSaveUI().ifPresent(svgFile -> {
    writeSVGToFile(svgDocument, svgFile);
});
```

That of course requires modifying `generateSaveUI`'s signature first. There are some great resources online if you look for them.